app/src/main/java/com/github/geohunt/app/ui/components/bounties/CreateNewBounty.kt

```diff
@@ -0,0 +1,168 @@
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Button
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.modifier.modifierLocalConsumer
+import androidx.compose.ui.platform.LocalUriHandler
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.github.geohunt.app.R
+import com.github.geohunt.app.i18n.DateFormatUtils
+import com.github.geohunt.app.model.Bounty
+import com.github.geohunt.app.model.Location
+import com.github.geohunt.app.ui.components.bounties.viewmodel.CreateBountyViewModel
+import com.github.geohunt.app.ui.components.challengecreation.*
+import com.github.geohunt.app.ui.components.utils.LocationPicker
+import com.ireward.htmlcompose.HtmlText
+import com.maxkeppeker.sheets.core.models.base.UseCaseState
+import com.maxkeppeler.sheets.calendar.CalendarDialog
+import com.maxkeppeler.sheets.calendar.models.CalendarSelection
+import java.time.LocalDate
+
+@Composable
+fun CreateNewBounty(
+    onFailure: (Throwable) -> Unit,
+    onSuccess: (Bounty) -> Unit,
+    viewModel: CreateBountyViewModel = viewModel(factory = CreateBountyViewModel.Factory)
+) {
+    val uriHandler = LocalUriHandler.current
+
+    Box(
+        modifier = Modifier.fillMaxSize(),
+        contentAlignment = Alignment.Center,
+    ) {
+        Column(
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally,
+            modifier = Modifier.verticalScroll(rememberScrollState())
+        ) {
+            Text(
+                text = "Create new bounty",
+                fontSize = 35.sp
+            )
+
+            Spacer(Modifier.height(55.dp))
+
+            val expirationDate = viewModel.expirationDate.collectAsState()
+            val startingDate = viewModel.startingDate.collectAsState()
+            val location = viewModel.location.collectAsState()
+            val name = viewModel.name.collectAsState()
+
+            BountySetting(
+                startingDate = startingDate.value,
+                expirationDate = expirationDate.value,
+                location = location.value,
+                name = name.value,
+                setStartingDate = viewModel::withStartingDate,
+                setExpirationDate = viewModel::withExpirationDate,
+                setLocation = viewModel::withLocation,
+                setName = viewModel::withName
+            )
+
+            Spacer(Modifier.height(15.dp))
+
+            HtmlText(

```

Comment:
I'm pretty sure this also was in the basic create challenge view. Maybe consider factorizing code